crates/sui-data-ingestion/src/workers/archival.rs

```diff
@@ -164,13 +164,15 @@ impl Worker for ArchivalWorker {
         let contents_blob = Blob::encode(&full_checkpoint_contents, BlobEncoding::Bcs)?;
         let blob_size = contents_blob.size();
         let summary_blob = Blob::encode(&checkpoint.checkpoint_summary, BlobEncoding::Bcs)?;
-        state.buffer.extend(contents_blob.data);
-        state.summary_buffer.extend(summary_blob.data);
+        contents_blob.write(&mut state.buffer)?;
+        summary_blob.write(&mut state.summary_buffer)?;
         state.checkpoint_range.end += 1;
 
         if !state.buffer.is_empty()
             && (((state.buffer.len() + blob_size) > self.commit_file_size)
                 || state.epoch != epoch
+                || (state.checkpoint_range.end - state.checkpoint_range.start)
+                    > (MAX_CHECKPOINTS_IN_PROGRESS / 2).try_into()?
```

Comment:
Could we add a comment about why this check is important?

Reply:
this one is for current backfill run. I won't land this check. Will think how to address this better